card/mod/core/set/all/name.rb

```diff
@@ -240,36 +240,9 @@ def repair_key
   end
 end
 
-event :rename, after: :set_name, on: :update do
-  if (existing_card = Card.find_by_key_and_trash(cardname.key, true)) &&
-     existing_card != self
-    existing_card.name = existing_card.name + "*trash"
-    existing_card.rename_without_callbacks
-    existing_card.save!
-  end
-end
-
-def suspend_name name
-  # move the current card out of the way, in case the new name will require
-  # re-creating a card with the current name, ie.  A -> A+B
-  Card.expire name
-  tmp_name = "tmp:" + UUID.new.generate
-  Card.where(id: id).update_all(name: tmp_name, key: tmp_name)
-end
-
-event :cascade_name_changes, :finalize, on: :update, changed: :name do
-  des = descendants
-  @descendants = nil # reset
-
-  des.each do |de|
-    # here we specifically want NOT to invoke recursive cascades on these
-    # cards, have to go this low level to avoid callbacks.
-    Rails.logger.info "cascading name: #{de.name}"
-    Card.expire de.name # old name
-    newname = de.cardname.replace name_before_last_save, name
-    Card.where(id: de.id).update_all name: newname.to_s, key: newname.key
-    de.update_referers = update_referers
-    de.refresh_references_in
-    Card.expire newname
-  end
+event :name_update_finished, :finalize, changed: :name, on: :save do
+  # The events to update references has to happen after :cascade_name_changes,
+  # but :cascade_name_changes is defined after the reference events and
+  # and additionaly it is defined on :update but some of the reference
+  # events are on :save. Hence we need this additional hook to organize these events.
```

Comment:
!

Reply:
woof.  so there are no known bugs associated with this?

Was the thinking before that we could figure out all the reference changes from parts?